aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -405,6 +405,10 @@ void _assert_async_cpu(const Tensor& self) {
   TORCH_CHECK(native::is_nonzero(self), "Expected Tensor with single nonzero value, but got zero");
 }
 
+void _assert_async_msg_cpu(const Tensor& self, c10::string_view assert_msg) {
+  TORCH_CHECK(native::is_nonzero(self), assert_msg != "" ? assert_msg : "Assertion is failed");
```

Comment:
Use AT_ASSERT

Reply:
This message is sus because you might violate semantics preserving notions here. 

You also don't ever pass the tensor to the message. Which is fine, but kinda lame.

Reply:
Sorry can you elaborate more what do you mean?